esphomeyaml/cookbook/bme280_environment.rst

```diff
@@ -0,0 +1,117 @@
+BME280 Environment
+==================
+
+.. seo::
+    :description: Instructions for setting up BME280 sensors in esphomelib and calculate altitude, absolute humidity, and sea level pressure.
+    :image: bme280.jpg
+    :keywords: BME280
+
+The :doc:`</esphomeyaml/components/sensor/bme280>` is a simple temperature, humidity, and pressure sensor with communication over I2C.
+With some simple math it is possible to either determine the height of the sensor, or the current pressure at sea level.
+This guide can be applied to any sensor measuring temperature and pressure at the same time, like the :doc:`</esphomeyaml/components/sensor/bmp280>`, or :doc:`</esphomeyaml/components/sensor/bme680>`.
+
+.. figure:: images/bme280-header.jpg
+    :align: center
+    :width: 75.0%
+
+The first step is to connect the sensor as described :doc:`here </esphomeyaml/components/switch/bme280>`.
+After validating the sensor is working, we can proceed and add some formulas.
+
+.. code-block:: yaml
+    globals:
+      - id: standard_sea_level_pressure_hpa
+        type: float
+        restore_state: no
+        initial_value: '1013.25'
+
+    sensor:
+      - platform: bme280
+        temperature:
+          name: "BME280 Temperature"
+          id: bme280_temperature
+        pressure:
+          name: "BME280 Pressure"
+          id: bme280_pressure
+        humidity:
+          name: "BME280 Humidity"
+          id: bme280_humidity
+        address: 0x77
+        update_interval: 15s
+      - platform: template
+        name: "Altitude"
+        lambda: >-
+          return ((id(bme280_temperature).state + 273.15) / 0.0065) * ((powf(id(standard_sea_level_pressure) / id(bme280_pressure).state), 0.190234) - 1);
+        update_interval: 15s
+      - platform: template
+        name: "Absolute Humidity"
+        lambda: >-
+          const float mw = 18.01534; 	// molar mass of water g/mol
+          const float r = 8.31447215; 	// Universal gas constant J/mol/K
+          return (6.112 * powf(2.718281828, (17.67 * id(bme280_temperature).state) / (id(bme280_temperature).state + 243.5)) * id(bme280_humidity).state * mw) / ((273.15 + id(bme280_temperature).state) * r);
+        update_interval: 15s
+
+Altitude and absolute humidity:
+-------------------------------
+The first block of code (``globals``) contains the variable standard sea level pressure in hPa, which you should fill in for your location.
+
+The second block ``sensor`` starts with the normal bme280 sensor components ``temperature``, ``pressure``, and ``humidity`` with each their own id.
+After the bme280 sensor, a :doc:`</esphomeyaml/components/sensor/template>` is defined to calculate the altitude in a lambda.
```

Comment:
`:doc:`/esphomeyaml/components/sensor/template` - remove the `<>`, otherwise the link text is empty.